src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,793 @@
+   */
+  @Prop() readonly resizable: boolean = false;
+  @Watch("resizable")
+  resizableChanged() {
+    // Schedule update for border size watcher
+    this.#checkBorderSizeWatcher = true;
+  }
+
+  /**
+   * Emitted when the dialog is opened.
+   */
+  @Event() dialogOpened: EventEmitter;
+
+  /**
+   * Emitted when the dialog is closed.
+   */
+  @Event() dialogClosed: EventEmitter;
+
+  connectedCallback() {
+    // Set RTL watcher
+    this.#rtlWatcher = new MutationObserver(() => {
+      this.#isRTLDirection = isRTL();
+
+      if (this.#isRTLDirection) {
+        this.el.style.setProperty(DIALOG_RTL, DIALOG_RTL_VALUE);
+        this.el.classList.add(DIALOG_RTL_CLASS);
+      } else {
+        this.el.style.removeProperty(DIALOG_RTL);
+        this.el.classList.remove(DIALOG_RTL_CLASS);
+      }
+    });
+
+    // Observe the dir attribute in the document
+    this.#rtlWatcher.observe(document.documentElement, {
+      attributeFilter: ["dir"]
+    });
+  }
+
+  componentWillRender() {
+    this.#checkPositionWatcher &&= false;
+
+    if (this.#checkBorderSizeWatcher) {
+      this.#checkBorderSizeWatcher = false;
+
+      // Wait until the resize edges have been rendered
+      requestAnimationFrame(() => {
+        this.#setBorderSizeWatcher();
+      });
+    }
+  }
+
+  componentDidLoad() {
+    if (!this.hidden) {
+      // Schedule update for watchers
+      this.#checkBorderSizeWatcher = true;
+      this.#checkPositionWatcher = true;
+      this.#showMethod();
+    }
+
+    // Initialize watchers
+    this.#setBorderSizeWatcher();
+  }
+
+  disconnectedCallback() {
+    // Defensive programming. Make sure the document does not have any unwanted handler
+    this.#handleDragEnd();
+    this.#removeBorderSizeWatcher();
+
+    // Disconnect RTL watcher
+    if (this.#rtlWatcher) {
+      this.#rtlWatcher.disconnect();
+      this.#rtlWatcher = null; // Free the memory
+    }
+  }
+
+  #addDraggingClass = () => {
+    if (!this.#dragging) {
+      this.#dialogRef.classList.add(DRAGGING_CLASS);
+      this.#dragging = true;
+    }
+  };
+
+  #removeDraggingClass = () => {
+    this.#dialogRef.classList.remove(DRAGGING_CLASS);
+    this.#dragging = false;
+  };
+
+  #handleDialogToggle = (event: ToggleEvent) => {
+    const willBeHidden = !(event.newState === "open");
+    this.hidden = willBeHidden;
+
+    // Emit events only when the action is committed by the user
+    if (willBeHidden) {
+      this.dialogClosed.emit();
+    } else {
+      this.dialogOpened.emit();
+    }
+  };
+
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  //                           Drag implementation
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  #handleMouseDown = (event: MouseEvent) => {
+    // We should not add preventDefault in this instance, because we would prevent some normal actions like clicking a button or focusing an input
```

Comment:
Limit comment length to the 80 column to keep the prettier style.

```suggestion
    // We should not add preventDefault in this instance, because we would prevent
    // some normal actions like clicking a button or focusing an input
```